pkg/rpm/resolver/resolver.go

```diff
@@ -0,0 +1,209 @@
+	"github.com/suse-edge/edge-image-builder/pkg/template"
+	"go.uber.org/zap"
+)
+
+const (
+	resolverImageRef = "pkg-resolver"
+	dockerfileName   = "Dockerfile"
+	rpmRepoName      = "rpm-repo"
+)
+
+//go:embed templates/Dockerfile.tpl
+var dockerfileTemplate string
+
+type Resolver struct {
+	// dir from where the resolver will work
+	dir string
+	// path to the image that the resolver will use as base
+	imgPath string
+	// type of the image that will be used as base (either ISO or RAW)
+	imgType string
+	// user provided packages for which dependency resolution will be done
+	packages *image.Packages
+	// user provided directory containing additional rpms for which dependency resolution will be done
+	customRPMDir string
+	// helper property, contains the names of the rpms that have been taken from the customRPMDir
+	rpms []string
+}
+
+// New creates a new Resolver instance that is based on the image context provided by the user
+func New(buildDir, imgConfDir string, imageDef *image.Definition) (*Resolver, error) {
+	rpmPath := filepath.Join(imgConfDir, "rpms")
+	if _, err := os.Stat(rpmPath); os.IsNotExist(err) {
+		rpmPath = ""
+	} else if err != nil {
+		return nil, fmt.Errorf("validating rpm dir exists: %w", err)
+	}
+
+	return &Resolver{
+		dir:          buildDir,
+		imgPath:      filepath.Join(imgConfDir, "images", imageDef.Image.BaseImage),
+		imgType:      imageDef.Image.ImageType,
+		packages:     &imageDef.OperatingSystem.Packages,
+		customRPMDir: rpmPath,
+	}, nil
+}
+
+// Resolve resolves all dependencies for the packages and third party rpms that have been configured by the user in the image context.
+// It then outputs the set of resolved rpms to a directory from which an RPM repository can be created. Returns the full path to the created
+// directory, the package/rpm names for which dependency resolution has been done, or an error if one has occurred.
+//
+// Parameters:
+//   - out - location where the RPM directory will be created
+func (r *Resolver) Resolve(out string, podman image.Podman) (rpmDir string, pkgList []string, err error) {
+	zap.L().Info("Resolving package dependencies...")
+
+	if err = r.buildBase(podman); err != nil {
+		return "", nil, fmt.Errorf("building base resolver image: %w", err)
+	}
+
+	if err = r.prepare(); err != nil {
+		return "", nil, fmt.Errorf("generating context for the resolver image: %w", err)
+	}
+
+	if err = podman.Build(r.generateBuildContextPath(), resolverImageRef); err != nil {
+		return "", nil, fmt.Errorf("building resolver image: %w", err)
+	}
+
+	id, err := podman.Create(resolverImageRef)
+	if err != nil {
+		return "", nil, fmt.Errorf("run container from resolver image %s: %w", resolverImageRef, err)
+	}
+
+	err = podman.Copy(id, r.generateRPMRepoPath(), out)
+	if err != nil {
+		return "", nil, fmt.Errorf("copying resolved package cache to %s: %w", out, err)
+	}
+
+	return filepath.Join(out, rpmRepoName), r.getPKGInstallList(), nil
```

Comment:
I would have to add a comment in the code for this logic, but essentially the flow is as follows:

1. Inside of the resolver image we create a directory with name `rpmRepoName` (passed to the Dockerfile [here](https://github.com/ipetrov117/edge-image-builder/blob/rpm-resolution/pkg/rpm/resolver/resolver.go#L135) and created [here](https://github.com/ipetrov117/edge-image-builder/blob/rpm-resolution/pkg/rpm/resolver/templates/Dockerfile.tpl#L31))
2. We resolve the pkg/rpm dependencies to this directory
3. We [copy](https://github.com/ipetrov117/edge-image-builder/blob/rpm-resolution/pkg/rpm/resolver/resolver.go#L85) this directory out of the container and return the path to the directory

After we have copied the directory on the EIB image, the resolver needs to return the path to this directory. Since the directory lies within the `out` dir, we concatenate the path - `filepath.Join(out, rpmRepoName)`. This is needed as sadly the podman copy logic does not allow for us to copy the contents of the directory, it instead copies the whole dir..

Reply:
Added a small comment in the code trying to explain why we concatenate the file path here.